crates/bevy_core_pipeline/src/skybox/node.rs

```diff
@@ -0,0 +1,98 @@
+use super::{SkyboxBindGroup, SkyboxMesh, SkyboxPipelineId};
+use bevy_ecs::{query::QueryState, world::World};
+use bevy_render::{
+    camera::ExtractedCamera,
+    mesh::GpuBufferInfo,
+    prelude::Mesh,
+    render_asset::RenderAssets,
+    render_graph::{Node, NodeRunError, RenderGraphContext},
+    render_resource::{
+        LoadOp, Operations, PipelineCache, RenderPassDepthStencilAttachment, RenderPassDescriptor,
+    },
+    renderer::RenderContext,
+    view::{ViewDepthTexture, ViewTarget, ViewUniformOffset},
+};
+
+pub(crate) struct SkyboxNode {
+    view_query: QueryState<(
+        &'static SkyboxPipelineId,
+        &'static SkyboxBindGroup,
+        &'static ViewUniformOffset,
+        &'static ExtractedCamera,
+        &'static ViewTarget,
+        &'static ViewDepthTexture,
+    )>,
+}
+
+impl Node for SkyboxNode {
+    fn run(
+        &self,
+        graph: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let Ok((pipeline_id, bind_group, view_uniform_offset, camera, target, depth))
+            = self.view_query.get_manual(world, graph.view_entity()) else { return Ok(()) };
+        let skybox_mesh = world.resource::<SkyboxMesh>();
+        let meshes = world.resource::<RenderAssets<Mesh>>();
+        let pipeline_cache = world.resource::<PipelineCache>();
+        let (Some(gpu_mesh), Some(pipeline)) = (
+            meshes.get(&skybox_mesh.handle),
+            pipeline_cache.get_render_pipeline(pipeline_id.0),
+        ) else { return Ok(()) };
+
+        {
+            let mut render_pass = render_context.begin_tracked_render_pass(RenderPassDescriptor {
```

Comment:
This shouldn't run in a separate pass, rather just at the end of the opaque/alpha mask render pass. How we do that exactly... giving it its own phase feels unnecessary. Maybe it can just directly look up what it needs from the view entity and execute the binding and draw commands explicitly on the pass at the end of the main opaque node.